samples/kvssink_gstreamer_sample.cpp

```diff
@@ -250,15 +270,38 @@ void determine_credentials(GstElement *kvssink, CustomData *data) {
 			"role-aliases", G_TYPE_STRING, role_alias, NULL);
 	
 	g_object_set(G_OBJECT (kvssink), "iot-certificate", iot_credentials, NULL);
-        gst_structure_free(iot_credentials);
+    gst_structure_free(iot_credentials);
     // kvssink will search for long term credentials in envvar automatically so no need to include here
     // if no long credentials or IoT credentials provided will look for credential file as last resort
     } else if(nullptr != (credential_path = getenv("AWS_CREDENTIAL_PATH"))){
         g_object_set(G_OBJECT (kvssink), "credential-path", credential_path, NULL);
     }
 }
 
-int gstreamer_live_source_init(int argc, char *argv[], CustomData *data, GstElement *pipeline) {
+// Function to put fragment metadata: name, value, and persist values
+// This is a sample function. Customers can write their own put_metadata and trigger it 
+// either using the timer or some other logic. This function can contain custom logic to
+// generate the metadata. To trigger the downstream event that calls the API putKinesisVideoFragmentMetadata,
+// put_fragment_metadata must be called as shown below.
+static void put_metadata(GstElement* element) {
+    std::ostringstream metadata_name_stream, metadata_value_stream;
+
+    metadata_name_stream << "metadata_name_" << data_global.metadata_counter;
+    metadata_value_stream << "metadata_value_" << data_global.metadata_counter;
+    
+    data_global.metadata_counter++;
+    data_global.persist_flag = !data_global.persist_flag;
+
+    if (!put_fragment_metadata(element, metadata_name_stream.str(), metadata_value_stream.str(), data_global.persist_flag)) {
```

Comment:
Want to confirm the logic here. We add metadata continuously until the queue fills up with persistent wetadata and then shut this timer down after. so after some time we will have always the same set of persistent metadata being added?

Reply:
Yes, because we wanted to make sure that toggling between the two works. Besides, this is a customizable function so the logic to generate the three parameters can be anything. I wanted to keep it simple and demonstrate that the logic works at the same time

Reply:
It will look bad to unfamiliar users if the error/warn log line is printed during runtime.

Since this is intentional behavior, can adjust to make it more clear that it isn't a true error or the error is expected.